### PR TITLE
refactor: refine event handler types and other minor inconsistencies

### PR DESCRIPTION
**Pull request by emmenko**
This is a follow up of the recent TS migration.

I noticed there were some minor inconsistencies or leftovers across several components so I went through them to clean things up.

### CODE REVIEW

File: packages/components/inputs/date-input/src/date-input.tsx [hunk 1]
```diff
@@ -279,7 +284,7 @@ const DateInput = (props: TDateInput) => {
                       clearSelection();
                     }
                     // ArrowDown
-                    if (event.keyCode === 40) {
+                    if (event.key === 'ArrowDown') {
```
Comment (CarlosCortizasCT) [inline packages/components/inputs/date-input/src/date-input.tsx:287]: I was wondering about our browsers version support policy as I read this [in MDN](https://devdocs.io/dom/keyboardevent/key/key_values):
image

Do we publicly state which browsers & versions we support?

File: packages/components/inputs/number-input/src/number-input.tsx [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { ChangeEventHandler, FocusEventHandler } from 'react';
+import { type ChangeEventHandler, type FocusEventHandler } from 'react';
```
Comment (CarlosCortizasCT) [inline packages/components/inputs/number-input/src/number-input.tsx:1]: nit:
```suggestion
import type { ChangeEventHandler, FocusEventHandler } from 'react';
```

File: packages/components/inputs/password-input/src/password-input.tsx [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { ChangeEventHandler, FocusEventHandler } from 'react';
+import { type ChangeEventHandler, type FocusEventHandler } from 'react';
```
Comment (CarlosCortizasCT) [inline packages/components/inputs/password-input/src/password-input.tsx:1]: nit:
```suggestion
import type { ChangeEventHandler, FocusEventHandler } from 'react';
```

File: packages/components/inputs/text-input/src/text-input.tsx [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import { FocusEventHandler, ChangeEventHandler } from 'react';
+import { type FocusEventHandler, type ChangeEventHandler } from 'react';
```
Comment (CarlosCortizasCT) [inline packages/components/inputs/text-input/src/text-input.tsx:1]: nit:
```suggestion
import type { FocusEventHandler, ChangeEventHandler } from 'react';
```

File: packages/components/inputs/date-input/src/date-input.tsx (hunk 1 above)
Comment (emmenko) [reply packages/components/inputs/date-input/src/date-input.tsx:287]: We have some info here

https://docs.commercetools.com/custom-applications/help-needed #browser-support

So it should be fine, as those mentioned there are pretty old versions

**Review by ddouglasz — APPROVED** [review]
LGTM

**Pull request merged**